src/CosmosEditorManager.ts

```diff
@@ -37,7 +44,17 @@ export class CosmosEditorManager {
         this._globalState = globalState;
     }
 
-    public async showDocument(editor: ICosmosEditor, fileName: string): Promise<void> {
+    public async showDocument(editor: ICosmosEditor, fileName: string, options: ShowEditorDocumentOptions): Promise<void> {
+        let column: vscode.ViewColumn = vscode.ViewColumn.Active;
+        let preserveFocus: boolean = false;
+        if (options && options.showInNextColumn) {
+            preserveFocus = true;
+            const activeEditor = vscode.window.activeTextEditor;
+            if (activeEditor && activeEditor.viewColumn >= vscode.ViewColumn.One) {
+                column = activeEditor.viewColumn < ViewColumn.Three ? activeEditor.viewColumn + 1 : ViewColumn.One;
```

Comment:
I see an application for using the modulo operator here :  `(activeEditor.viewColumn%3) + 1`